flow/raster_cache.h

```diff
@@ -143,9 +148,12 @@ class RasterCache {
   // if the item was disabled due to conditions discovered during |Preroll|
   // or if the attempt to populate the entry failed due to bounds overflow
   // conditions.
+  // If |is_root_canvas| is false, the raster cache will preserve the original
+  // RTree of cached content.
   bool Draw(const RasterCacheKeyID& id,
             DlCanvas& canvas,
-            const DlPaint* paint) const;
+            const DlPaint* paint,
+            bool is_root_canvas = true) const;
```

Comment:
Perhaps at this level the flag should be named something more specific to its duties. Whether or not there is a root canvas in the system is outside of the context of these classes. So, while `is_root_canvas` makes sense at the layer painting level, here it should be something like `use_rtree` or `preserve_rtree` (which would invert its value) or `ignore_rtree` (which would have the same value as "is root", but feels more awkward of a name). These classes don't care if there is anything about a root canvas, they just want to know if they should use the supplied rtree or not...

The other flag used in the layers could be named something like `rendering_to_platform_view` to make it compatible with `preserve_rtree` here without inverting it?

Reply:
I like `preserve_rtree`. I think the flag in `PaintContext` could be `is_overlay_layer` or `rendering_above_platform_view`.

Reply:
Went with `rendering_above_platform_view`. To me it communicates the z-index clearer, but it's probably subjective.

Reply:
My bad. Pre-push check failed and I didn't notice that the push didn't get through.